controllers/datamovement_controller_test.go

```diff
@@ -21,35 +21,85 @@ package controllers
 
 import (
 	"context"
+	_ "embed"
+	"fmt"
 	"os"
+	"path/filepath"
+	"time"
 
 	nnfv1alpha1 "github.com/NearNodeFlash/nnf-sos/api/v1alpha1"
+	"github.com/google/uuid"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	. "github.com/onsi/gomega/gstruct"
 
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// This is dumped into a temporary file and then ran as a bash script.
+//
+//go:embed fakeProgressOutput.sh
+var progressOutputScript string
+
 var _ = Describe("Data Movement Test" /*Ordered, (Ginkgo v2)*/, func() {
-	var dm *nnfv1alpha1.NnfDataMovement = nil
-	var srcPath, destPath string
+	var dm *nnfv1alpha1.NnfDataMovement
+	var cm *corev1.ConfigMap
+	var cmData = make(map[string]string)
+	createCm := true
+	var tmpDir string
+	var srcPath string
+	var destPath string
+	dmCancel := false
 
-	BeforeEach(func() {
-		var err error
+	createFakeProgressScript := func() string {
+		f, err := os.CreateTemp(tmpDir, "fakeProgressOutput-*.sh")
+		Expect(err).ToNot(HaveOccurred())
+		filePath := f.Name()
 
-		srcPath, err = os.MkdirTemp("/tmp", "dm-test")
+		bytes := []byte(progressOutputScript)
+		_, err = f.Write(bytes)
 		Expect(err).ToNot(HaveOccurred())
 
-		destPath, err = os.MkdirTemp("/tmp", "dm-test")
+		err = os.Chmod(filePath, 0755)
 		Expect(err).ToNot(HaveOccurred())
 
+		return filePath
+	}
+
+	BeforeEach(func() {
+		var err error
+		tmpDir, err = os.MkdirTemp("/tmp", "dm-test")
+		Expect(err).ToNot(HaveOccurred())
+
+		srcPath = filepath.Join(tmpDir, "src")
+		destPath = filepath.Join(tmpDir, "dest")
+
+		// Set default values
+		dmCancel = false
+		createCm = true
+		cmData[configMapKeyCmd] = "sleep 1"
+		cmData[configMapKeyProgInterval] = "1s"
+		cmData[configMapKeyDcpProgInterval] = "1"
+		cmData[configMapKeyNumProcesses] = ""
+		os.Unsetenv("NNF_NODE_NAME")
```

Comment:
Do what you can to keep these things local. In this scenario you only have one test that is ever setting NNF_NODE_NAME, so why not unset it in that one test.